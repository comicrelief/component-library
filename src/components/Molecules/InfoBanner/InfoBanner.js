import React from 'react';
import styled from 'styled-components';
import Moment from 'react-moment';
import CurrencyFormat from "react-currency-format";

const Container = styled.section`
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    flex-direction: column;
    position: relative;
    color: white;
    padding: 60px 0;
    background: ${p => p.background ? p.background : "transparent"};
    @media (min-width: 700px) {
        flex-direction: row;
    }
`;

const Info = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    margin-bottom: 40px;
    flex-direction: column;
    @media (min-width: 700px) {
        width: calc(100%/3);
    }
`;

const Label = styled.h3`
    text-transform: uppercase;
    font-size: 16px;
`;

const Desc = styled.p`
    font-size: 27px;
    font-weight: bold;
`;

/**
 * Info Banner for landing pages.
 */

const InfoBanner = ({title, background, theme, startDate, endDate, country, amount}) => {
    return (
        <Container background={background}>
            <Info>
                <Label>Project Name</Label>
                <Desc>{title}</Desc>
            </Info>
            <Info>
                <Label>End Date</Label>
                <Desc>
                    <Moment format="MMMM Do YYYY">{endDate}</Moment>
                </Desc>
            </Info>
            <Info>
                <Label>Funding theme</Label>
                <Desc>{theme}</Desc>
            </Info>
            <Info>
                <Label>Amount Awarded</Label>
                <Desc>
                    <CurrencyFormat value={amount} displayType={'text'} thousandSeparator={true} prefix={'Â£'} renderText={value => value} />
                </Desc>
            </Info>            
            <Info>
                <Label>Start Date</Label>
                <Desc>
                    <Moment format="MMMM Do YYYY">{startDate}</Moment>
                </Desc>
            </Info>
            <Info>
                <Label>Beneficiary Country</Label>
                <Desc>{country}</Desc>
            </Info>
        </Container>
    );
};

export default InfoBanner;