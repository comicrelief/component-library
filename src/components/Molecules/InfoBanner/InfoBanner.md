Info Banner Example:

```js
const title = 'Hello There';
const amount = 'Â£ 2,000,000';
const theme = 'Be happy';
const startDate = '1st June 2010';
const endDate = '1st June 2020';
const country = 'Uk';

<InfoBanner
  title={title}
  amount={amount}
  theme={theme}
  startDate={startDate}
  endDate={endDate}
  country={country}
/>;
```
