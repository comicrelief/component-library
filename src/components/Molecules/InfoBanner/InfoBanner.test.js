import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import InfoBanner from './InfoBanner';

it('renders correctly', () => {
  const title = 'Hello There';
  const amount = 'Â£ 2,000,000';
  const theme = 'Be happy';
  const startDate = '1st June 2010';
  const endDate = '1st June 2020';
  const country = 'Uk';
  const tree = renderWithTheme(
    <InfoBanner
      background="#2C0230"
      title={title}
      amount={amount}
      theme={theme}
      startDate={startDate}
      endDate={endDate}
      country={country}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
