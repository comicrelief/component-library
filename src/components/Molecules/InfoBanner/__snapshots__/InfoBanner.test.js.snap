// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  text-transform: uppercase;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c3 {
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c4 {
  position: relative;
  display: inline;
  line-height: 1.5rem;
  color: #000000;
  font-weight: normal;
}

.c4:hover,
.c4:focus {
  color: #000000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: 100%;
  margin-bottom: 2rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:740px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:740px) {
  .c1 {
    width: calc(100% / 3);
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <h3
      className="c2"
      color="inherit"
      size="md"
    >
      Project Name
    </h3>
    <p
      className="c3"
      color="inherit"
      size="s"
    >
      Hello There
    </p>
  </div>
  <div
    className="c1"
  >
    <h3
      className="c2"
      color="inherit"
      size="md"
    >
      End Date
    </h3>
    <p
      className="c3"
      color="inherit"
      size="s"
    >
      1st June 2020
    </p>
  </div>
  <div
    className="c1"
  >
    <h3
      className="c2"
      color="inherit"
      size="md"
    >
      Funding theme
    </h3>
    <p
      className="c3"
      color="inherit"
      size="s"
    >
      <a
        className="c4"
        color="red"
        href="cr.com"
        target="_self"
        type="standard"
      >
        Be happy
      </a>
    </p>
  </div>
  <div
    className="c1"
  >
    <h3
      className="c2"
      color="inherit"
      size="md"
    >
      Amount Awarded
    </h3>
    <p
      className="c3"
      color="inherit"
      size="s"
    >
      Â£2,000,000
    </p>
  </div>
  <div
    className="c1"
  >
    <h3
      className="c2"
      color="inherit"
      size="md"
    >
      Start Date
    </h3>
    <p
      className="c3"
      color="inherit"
      size="s"
    >
      1st June 2010
    </p>
  </div>
  <div
    className="c1"
  >
    <h3
      className="c2"
      color="inherit"
      size="md"
    >
      Beneficiary Country
    </h3>
    <p
      className="c3"
      color="inherit"
      size="s"
    >
      Uk
    </p>
  </div>
</section>
`;
